### PR TITLE
feat: flesh out signals demo

### PR DESCRIPTION
**Pull request by DaveMBush**
# Issue Number: #900 

# Body

The signal demo is not feature parity with the classic ngrx demo

## Summary by CodeRabbit

- **New Features**
  - Introduced three new routes: treeNoRefresh, treeNoRemove, and treeNoDirty, each with dedicated tree components and tailored reactive state management.
  - Added standalone tree components supporting reactive location and department data, with event handling for location changes.
  - Enhanced state selectors and entity definitions for locations, departments, and department children, enabling hierarchical data handling with different mark-and-delete policies.
  - Implemented styles ensuring tree components occupy full container height.

- **Bug Fixes**
  - None.

- **Tests**
  - Added comprehensive test suites for current location stores and location selectors, covering multiple scenarios and edge cases.

- **Documentation**
  - None.

### CODE REVIEW

File: apps/demo-ngrx-signals/src/app/routes/tree-no-remove/feature.const.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const featureName = 'tree-standard';
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/demo-ngrx-signals/src/app/routes/tree-no-remove/feature.const.ts]: this should be fixed now

File: apps/demo-ngrx-signals/src/app/routes/tree-no-refresh/feature.const.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export const featureName = 'tree-standard';
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/demo-ngrx-signals/src/app/routes/tree-no-refresh/feature.const.ts]: this should be fixed now

File: apps/demo-ngrx-signals/src/app/routes/tree-no-dirty/store/department/select-departments.selector.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { createSmartSignal } from '@smarttools/smart-signals';
+
+import { Department } from '../../../../shared/department/department.interface';
+import { featureName } from '../../feature.const';
+
+export const selectDepartments = createSmartSignal<Department>(
+  featureName,
+  'departments',
+);
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/demo-ngrx-signals/src/app/routes/tree-no-dirty/store/department/select-departments.selector.ts:9]: this should be fixed now

File: apps/demo-ngrx-signals/src/app/routes/tree-no-refresh/store/department/select-departments.selector.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { createSmartSignal } from '@smarttools/smart-signals';
+
+import { Department } from '../../../../shared/department/department.interface';
+import { featureName } from '../../feature.const';
+
+export const selectDepartments = createSmartSignal<Department>(
+  featureName,
+  'departments',
+);
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/demo-ngrx-signals/src/app/routes/tree-no-refresh/store/department/select-departments.selector.ts:9]: this should be fixed now

File: apps/demo-ngrx-signals/src/app/routes/tree-no-dirty/store/department-children/department-child.selector.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { createSmartSignal } from '@smarttools/smart-signals';
+
+import { DepartmentChild } from '../../../../shared/department-children/department-child.interface';
+import { featureName } from '../../feature.const';
+
+export const selectDepartmentChildren = createSmartSignal<DepartmentChild>(
+  featureName,
+  'departmentChildren',
+);
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/demo-ngrx-signals/src/app/routes/tree-no-dirty/store/department-children/department-child.selector.ts:9]: there's nothing to test here.  I only focus on branch logic

File: apps/demo-ngrx-signals/src/app/routes/tree-no-refresh/store/department/select-departments-children.selector.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+// jscpd:ignore-start
+// intentionally duplicated.
+import { createSmartSignal } from '@smarttools/smart-signals';
+
+import { featureName } from '../../feature.const';
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/demo-ngrx-signals/src/app/routes/tree-no-refresh/store/department/select-departments-children.selector.ts:5]: this should be fixed now

File: apps/demo-ngrx-signals/src/app/routes/tree-no-remove/store/locations/selectors/select-locations-departments.selectors.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      childFeature: featureName,
+      childEntity: 'departments',
+      parentField: 'departments',
+      parentFeature: featureName,
+      parentEntity: 'locations',
+      childSelector: selectDepartmentsChildren,
+    },
+  ],
+);
+// jscpd:ignore-end
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/demo-ngrx-signals/src/app/routes/tree-no-remove/store/locations/selectors/select-locations-departments.selectors.ts:22]: this should be fixed now

File: apps/demo-ngrx-signals/src/app/routes/tree-no-refresh/store/top/select-top-locations.selectors.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
+// jscpd:ignore-start
+// intentionally duplicated.
+import { createSmartSignal } from '@smarttools/smart-signals';
+
+import { featureName } from '../../feature.const';
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/demo-ngrx-signals/src/app/routes/tree-no-refresh/store/top/select-top-locations.selectors.ts:5]: this should be fixed now

File: apps/demo-ngrx-signals/src/app/routes/tree-no-remove/store/top/select-top-locations.selectors.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+// jscpd:ignore-start
+// intentionally duplicated.
+import { createSmartSignal } from '@smarttools/smart-signals';
+
+import { featureName } from '../../feature.const';
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/demo-ngrx-signals/src/app/routes/tree-no-remove/store/top/select-top-locations.selectors.ts:5]: this should be fixed now

File: apps/demo-ngrx-signals/src/app/routes/tree-no-refresh/feature.const.ts [hunk 10]
```diff
@@ -0,0 +1 @@
+export const featureName = 'tree-no-refresh';
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/demo-ngrx-signals/src/app/routes/tree-no-refresh/feature.const.ts:1]: this should be fixed now

File: apps/demo-ngrx-signals/src/app/routes/tree-no-dirty/feature.const.ts [hunk 11]
```diff
@@ -0,0 +1 @@
+export const featureName = 'tree-no-refresh';
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/demo-ngrx-signals/src/app/routes/tree-no-dirty/feature.const.ts:1]: this should be fixed now
